src/steps/intune/steps/noncomplianceFindings/__tests__/index.test.ts

```diff
@@ -0,0 +1,80 @@
+import {
+  createMockStepExecutionContext,
+  Recording,
+} from '@jupiterone/integration-sdk-testing';
+import { setupAzureRecording } from '../../../../../../test/recording';
+import { config } from '../../../../../../test/config';
+import { fetchNonComplianceFindings } from '..';
+import { fetchDevices } from '../../devices';
+import { fetchDeviceConfigurations } from '../../deviceConfigurations';
+import { entities } from '../../../constants';
+
+let recording: Recording;
+
+afterEach(async () => {
+  if (recording) {
+    await recording.stop();
+  }
+});
+
+describe('fetchNonComplianceFindings', () => {
+  test('should make entities and relationships correctly', async () => {
+    recording = setupAzureRecording({
+      directory: __dirname,
+      name: 'fetchNonComplianceFindings',
+    });
+    const context = createMockStepExecutionContext({ instanceConfig: config });
+
+    await fetchDevices(context);
+    await fetchDeviceConfigurations(context);
+    await fetchNonComplianceFindings(context);
+
+    const deviceConfiguration = context.jobState.collectedEntities.filter(
+      (e) => {
+        return e._class.includes('Finding');
+      },
+    );
```

Comment:
How come this test only validates `deviceConfiguration`s? I think this is supposed to be for `noncomplianceFindings`, but it doesn't do any validation for them.

It would also be great to make sure that you know the entire set of collected entities you expect in this test. E.g. once you filter out `device`s, `deviceConfigurations`, and `noncomplianceFindings`, be sure that the `context.jobState.collectedEntities` is empty.

Another thing you could have done for this test was to add some dummy entities for `device` and `deviceConfiguration` to your `mockStepExecutionContext`. 

```typescript
      const context = createMockStepExecutionContext({ 
      instanceConfig: config,
      entities: [
        {
          // device entity that you'll want to create relationships from
        },
        {
          // deviceConfiguration entity that you'll want to create relationships from
        }
      ]
    });
```

That way, you won't have those entities show up in your jobs state, and you will be sure that `context.jobState.collectedEntities` contains _only_ `noncomplianceFinding` entities.

```typescript
expect(context.jobState.collectedEntities).toMatchGraphObjectSchema({
  _class: entities.NONCOMPLIANCE_FINDING._class
});

Reply:
Ah. I see now that line 32 may be a copy-paste error. It should be 

```suggestion
    const noncomplianceFindingEntities = context.jobState.collectedEntities.filter(
      (e) => {
        return e._class.includes('Finding');
      },
    );
```

Reply:
Ah. I see now that line 32 may be a copy-paste error. It should be 

```suggestion
    const noncomplianceFindingEntities = context.jobState.collectedEntities.filter(
      (e) => {
        return e._class.includes('Finding');
      },
    );
```

Reply:
Yeah, didn't change the variable. I'll update that and add the more explicit tests.

- [ ] fix noncomplianceFindingEntities variable name
- [ ] add a test that filters out devices, deviceConfigurations, and noncomplianceFindings, and checks that the context.jobState.collectedEntities is empty.